### PR TITLE
Fixed timestamp conversion on Windows

### PR DESCRIPTION
**Pull request by alecuba16**
Fixed timestamp conversion on windows systems as indicated here:

https://github.com/apache/spark/pull/36566/files/d47a175f35339814e7d952ededaac63d85a15834

### What changes were proposed in this pull request?

Fix the timestamp conversion on pre 1970 dates in windows systems, only two functions have been modified.

### Why are the changes needed?

This changes are needed to fix the bug of parse pre 1970 dates as indicated in [this pull request for master](https://github.com/apache/spark/pull/36566). In this case is required, because AWS GLUE spark serverless service uses spark 3.1.1 (branch 3.1).

### Does this PR introduce _any_ user-facing change?

No

### CODE REVIEW

**Comment by itholic** [comment]
Could you file a JIRA for [Apache Spark JIRA](https://issues.apache.org/jira/projects/SPARK/issues/SPARK-34827?filter=allopenissues) and add it to a title ??

You can also refer to [contribution guide for Apache Spark](https://spark.apache.org/contributing.html).

File: python/pyspark/sql/types.py [hunk 1]
```diff
@@ -176,14 +177,36 @@ def needConversion(self):
 
     def toInternal(self, dt):
         if dt is not None:
-            seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
-                       else time.mktime(dt.timetuple()))
+            if platform.system().lower() == "windows":
+                """
+                On Windows, the current value is converted to a timestamp
+                when the current value is less than 1970
+                """
```
Comment (itholic) [inline python/pyspark/sql/types.py:184]: docstring should be placed at the top of the function.

I think we can just leave the multi-line comments in this case.

File: python/pyspark/sql/types.py [hunk 2]
```diff
@@ -498,20 +521,20 @@ def __init__(self, fields=None):
 
     def add(self, field, data_type=None, nullable=True, metadata=None):
         """
-        Construct a :class:`StructType` by adding new elements to it, to define the schema.
+        Construct a StructType by adding new elements to it, to define the schema.
```
Comment (itholic) [inline python/pyspark/sql/types.py:524]: qq: Why do we change this ?

Comment (alecuba16) [reply python/pyspark/sql/types.py:524]: Not needed, seems to be something that have been changed when I was forking the branch 3.1. It should be the same as the current 3.1. I will fix that.

**Pull request closed without merging**